analysis/ic50/__init__.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+from scipy.optimize import curve_fit
+from sklearn.metrics import mean_squared_error
+from sklearn.preprocessing import MinMaxScaler
+
+
+def _sigmoid(x: Union[np.array, float], a: float, b: float, c: float, d: float) -> Union[np.array, float]:
+    return ((a - d) / (1.0 + ((x / c) ** b))) + d
+
+
+def _get_starting_parameters(concentrations: np.array, scaled_measurements: np.array) -> List[float]:
+    a = scaled_measurements[0]
+    d = scaled_measurements[-1]
+    b = 1
+    c = concentrations[(np.abs(scaled_measurements - (a + d) / 2)).argmin()]
+    return [a, b, c, d]
+
+
+def _fit_sigmoid(
+        measurements: np.array,
+        scaler: MinMaxScaler, fit_scaler: bool = True,
+) -> Tuple[np.array, np.array, float, MinMaxScaler]:
+    if fit_scaler:
+        scaled_signal = scaler.fit_transform(measurements[:, 1].reshape(-1, 1)).reshape(-1)
+    else:
+        scaled_signal = scaler.transform(measurements[:, 1].reshape(-1, 1)).reshape(-1)
+
+    starting_point = _get_starting_parameters(measurements[:, 0], scaled_signal)
+    try:
+        with np.errstate(invalid='ignore'):
+            fit_param = curve_fit(_sigmoid, measurements[:, 0], np.array(scaled_signal),
+                                  p0=starting_point)
+    except RuntimeError:
```

Comment:
Why don't we catch `ValueError` or `OptimizeWarning`? As I can see they are also possible in `curve_fit`

Reply:
I added a check for NaN's in data so `ValueError` will not be possible. Regarding `OptimizeWarning` I haven't encountered it, but it's not an exception so doesn't break the code. I can either let it be (so printed as `stderr`) or simply suppress it (as with `np.errstate`) because it carries little information and clogs the output logs.